### ISSUE TITLE
Take Screenshot doesn't process arguments correctly when run on failure

### ISSUE THREAD

**Issue by tomekTieto** [MarketSquare/robotframework-browser#1356]
**Bug description**
Take Screenshot doesn't process arguments correctly when it's registered to be run on failure. Normally when called explicitly it's working fine. Because of that as example unable to take fullPage screenshot.

**Steps To Reproduce**
Look at my 2 code snippets below:
```
Take Screenshot not working correctly when registered to run on failure
    [Setup]  New Page  https://www.yahoo.com/
    Browser.Register Keyword To Run On Failure  Take Screenshot  fullPage=True
    Click  //button[@name="agree"]
    Wait Until Network Is Idle
    Wait For Elements State  ddd  state=visible
    [Teardown]  Close Page
```

```
Take Screenshot working correctly when explicitly called
   [Setup]  New Page  https://www.yahoo.com/
    Click  //button[@name="agree"]
    Wait Until Network Is Idle
    Take Screenshot  fullPage=True
    [Teardown]  Close Page
```

**Expected behavior**
Take Screenshot should take full page screenshot for both examples.

**Screenshots**
Incorrect - screenshot taken from viewport as default. Notice that filename was taken from the argument fullPage=True
![incorrect](https://user-images.githubusercontent.com/39945193/135421206-4af90544-1d13-48f5-9ce6-c5d421c53821.png)
![incorrect_argument_as_filename](https://user-images.githubusercontent.com/39945193/135421282-7c79fe1f-b0fc-45fb-850f-b16dbfdfadbe.png)

Correct - full page screenshot. File name correct according to default filename argument
![correct](https://user-images.githubusercontent.com/39945193/135421810-f6e63d91-df22-40ae-a68d-16ac1c521c8e.png)
![correct_filename](https://user-images.githubusercontent.com/39945193/135421834-166ce2d7-75c0-4277-a097-ace8970532d3.png)

**Desktop:**
 - OS: Windows 10 / docker image rfbrowser-stable:v8.0.1
 - Browser Chrome
 - Version 94.0.4606.61

**Additional context**
Checked other KWs to be registered on failure with arguments like Log and they're working correctly.

**Comment by aaltat** [comment]
@all-contributors please add @tomekTieto for bugs

**Comment by aaltat** [comment]
This is true, we do not do conversion, we just take arguments as they are, which in this case is plain strings and feed them forward. When we run keyword, we did expect that Robot Framework has done argument conversion and do not check any of the arguments.

The problem is that if there is keyword which is not keyword from this library, we do know the arguments types and therefore we can not do argument conversion. But it might be possible to do argument conversion for keywords which are from this library.

**Comment by aaltat** [comment]
Reading the code also reveals nice bug in our [code](https://github.com/MarketSquare/robotframework-browser/blob/main/Browser/browser.py#L896-L899) 
```
            if is_same_keyword(self.run_on_failure_keyword["name"], "Take Screenshot"):
                args = self.run_on_failure_keyword["args"]
                path = args[0] if args else self._failure_screenshot_path()
                self.take_screenshot(path)
```
When `Take Screenshot` is run, code puts `filename` whatever was given as first argument. That is also wrong.

**Comment by aaltat** [comment]
Even with that bug fixed we need to conversion for the arguments internally. Library relies on RF to the conversion when methods are run as keywords but for this usage case we need to do the conversion internally. Will look some RF api to do it, because argument conversion is pretty complex problem.

**Comment by aaltat** [comment]
Had discussion with Pekke on Slack and this should work nicely:
```
>>> from robot.running.arguments import PythonArgumentParser
>>> def kw(a: int):
...     pass
... 
>>> spec = PythonArgumentParser().parse(kw)
>>> spec.resolve(['42'])
([42], [])
>>> spec.resolve(['a=42'])
([], [('a', 42)])
>>> spec.resolve(['oops'])
Traceback (most recent call last):
[snip]
ValueError: Argument 'a' got value 'oops' that cannot be converted to integer.
```

**Comment by tomekTieto** [comment]
Added even a debug log but nothing special there regarding issue
[debug-20211015-124501.log](https://github.com/MarketSquare/robotframework-browser/files/7352877/debug-20211015-124501.log)

**Comment by aaltat** [comment]
As a workaround doing something like this should work 
https://github.com/MarketSquare/robotframework-browser/blob/main/atest/test/01_Browser_Management/run_on_failure.robot#L47

But there definitely is a bug.

**Comment by tomekTieto** [comment]
Yes, using positional arguments instead of named arguments is working fine

**Comment by aaltat** [comment]
Lets move the discussion in #1396
